x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/action/TransportStartDataFrameAnalyticsAction.java

```diff
@@ -281,12 +281,23 @@ protected DiscoveryNode selectLeastLoadedNode(ClusterState clusterState, Predica
         protected void nodeOperation(AllocatedPersistentTask task, StartDataFrameAnalyticsAction.TaskParams params,
                                      PersistentTaskState state) {
             LOGGER.info("[{}] Starting data frame analytics", params.getId());
-            DataFrameAnalyticsTaskState startedState = new DataFrameAnalyticsTaskState(DataFrameAnalyticsState.STARTED,
-                task.getAllocationId());
-            task.updatePersistentTaskState(startedState, ActionListener.wrap(
-                response -> manager.execute((DataFrameAnalyticsTask) task),
-                task::markAsFailed
-            ));
+            DataFrameAnalyticsTaskState analyticsTaskState = (DataFrameAnalyticsTaskState) state;
+
+            // If we are "stopping" there is nothing to do
+            if (analyticsTaskState != null && analyticsTaskState.getState() == DataFrameAnalyticsState.STOPPING) {
```

Comment:
from what I can tell, nothing is setting the state to `STOPPING`. However, it makes sense to me that we should not attempt to execute if we are in the middle of stopping.

Reply:
> We might have to stop the persistent task so that it actually goes to `stopped`.

The problem if you do this is that all record of the fact that the task was forcibly interrupted while stopping is lost, which is not good from a support point-of-view.

If we think we might ever need a `stopping` state then I agree it's good for it to be in the enum from day 1 to make BWC easier.  I think the code here is OK as it will leave `stopping` tasks that are reassigned in limbo.  If we ever write some code that puts a task into `stopping` state then in the same version we'll need to add some sort of force-stop endpoint that can get it out of that state.

Something for a different PR is that we should probably add a `failed` state too.  At the moment I guess if the analytics process crashes we delete its persistent task.  But that makes debugging in a large cluster very hard - see https://github.com/elastic/elasticsearch/issues/30006#issuecomment-448235425.  A `failed` state would provide a route to discover which node the process failed on from a support diag.  Then `failed` would be added to the `if` statement here alongside `stopping`.

Reply:
++

I'll raise an issue for handling `failed` state.

Reply:
++

`failed` state is definitely useful, especially since we can add cause information in the clusterstate and provide that to the user.